app/lib/router/shell_routers/chat_shell_router.dart

```diff
@@ -1,16 +1,35 @@
 import 'package:acter/common/providers/chat_providers.dart';
 import 'package:acter/common/utils/routes.dart';
 import 'package:acter/common/utils/utils.dart';
-import 'package:acter/features/chat/chat-ng/pages/ng_profile_page.dart';
-import 'package:acter/features/chat/chat-ng/pages/room_page_ng.dart';
 import 'package:acter/features/chat/pages/room_page.dart';
 import 'package:acter/features/chat/pages/room_profile_page.dart';
 import 'package:acter/features/chat/widgets/chat_layout_builder.dart';
+import 'package:acter/features/chat_ng/widgets/rooms_list.dart';
 import 'package:acter/features/invite_members/pages/invite_page.dart';
+import 'package:acter/features/settings/providers/settings_providers.dart';
 import 'package:acter/features/space/settings/pages/visibility_accessibility_page.dart';
 import 'package:acter/router/router.dart';
+import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
 
+/// define the chat layout builder depending on whether the user has set
+/// the chat-ng feature.
+Widget _chatLayoutBuilder({Widget? centerChild, Widget? expandedChild}) {
+  final isChatNg =
+      rootNavKey.currentContext?.read(isActiveProvider(LabsFeature.chatNG)) ==
+          true;
+  return isChatNg
```

Comment:
this is the actual core of the change: do the `isActivateProvider(LabsFeature.chatNG)` check within the GoRoute-Builder and pick the right Widget accordingly.